src/kv/LockResolver.cc

```diff
@@ -245,11 +263,186 @@ void LockResolver::resolvePessimisticLock(Backoffer & bo, LockPtr lock, std::uno
     }
 }
 
+void LockResolver::resolveLockAsync(Backoffer & bo, LockPtr lock, TxnStatus & status)
+{
+    AsyncResolveDataPtr resolve_data{};
+    try
+    {
+        resolve_data = checkAllSecondaries(bo, lock, status);
+    }
+    catch (Exception & e)
+    {
+        throw;
+    }
+
+    status.commit_ts = resolve_data->commit_ts;
+
+    resolve_data->keys.push_back(lock->primary);
+    auto [keys_by_region, region_id] = cluster->region_cache->groupKeysByRegion(bo, resolve_data->keys);
+    (void)region_id;
+    std::vector<std::thread> threads;
+    std::atomic<int> errors{};
+    for (auto & pair : keys_by_region)
+    {
+        auto & region_id = pair.first;
+        auto & locks = pair.second;
+        threads.emplace_back([&]() {
+            try
+            {
+                resolveRegionLocks(bo, lock, region_id, locks, status);
+            }
+            catch (Exception & e)
+            {
+                errors.fetch_add(1);
+                log->error("resolveRegionLocks error: " + e.displayText());
+            }
+        });
+    }
+
+    for (auto & t : threads)
+    {
+        t.join();
+    }
+
+    if (errors.load() > 0)
+    {
+        throw Exception("AsyncCommit recovery finished with errors", ErrorCodes::UnknownError);
```

Comment:
I bet the errorcodes is not "unknown"

Reply:
Yes, we should log all of the errors. But it seems we don't have a proper way to collect error codes thrown by worker threads, do you have any idea?